Lib/bdb.py

```diff
@@ -33,6 +33,7 @@ def __init__(self, skip=None):
         self.breaks = {}
         self.fncache = {}
         self.frame_returning = None
+        self.botframe = None
```

Comment:
Not that it's a huge deal, but now this attribute needs to get set twice, once on init and once on reset. It would be nice if that didn't have to be the case.

Reply:
It is probably appropriate to just call reset() from __init__, but I tend to fix bugs with the minimum necessary change and then do refactors in a separate commit.